legend-depot-store-metrics/src/main/java/org/finos/legend/depot/store/metrics/services/InMemoryQueryMetricsRegistry.java

```diff
@@ -0,0 +1,60 @@
+//  Copyright 2021 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+package org.finos.legend.depot.store.metrics.services;
+
+import org.finos.legend.depot.store.admin.domain.metrics.VersionQueryMetric;
+import org.finos.legend.depot.store.metrics.api.QueryMetricsRegistry;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Date;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Singleton
+public class InMemoryQueryMetricsRegistry implements QueryMetricsRegistry
+{
+    private ConcurrentLinkedQueue<VersionQueryMetric> metrics;
+
+    @Inject
+    public InMemoryQueryMetricsRegistry()
+    {
+        this.metrics = new ConcurrentLinkedQueue<>();
+    }
+
+    public void record(String groupId, String artifactId, String versionId)
+    {
+        metrics.offer(new VersionQueryMetric(groupId, artifactId, versionId));
+    }
+
+    public void record(String groupId, String artifactId, String versionId, Date date)
```

Comment:
why do we need a record with a date.? if we do you should implement the above based on this one